pages/recording/upload.tsx

```diff
@@ -21,9 +23,17 @@ function UploadScreenWrapper({ onUpload }: { onUpload: () => void }) {
   if (loading) {
     return <LoadingScreen message="Loading settings..." />;
   }
+  if (loading2) {
+    return <LoadingScreen message="Loading team info..." />;
+  }
```

Comment:
nit: I don't think we need to show separate user-facing messages for this.
```suggestion
  if (userSettingsLoading || pendingWorkspacesLoading) {
    return <LoadingScreen message="Loading..." />;
  }
```